packages/api/es/esFileSearchQueue.js

```diff
@@ -0,0 +1,131 @@
+'use strict';
+
+const flatten = require('lodash/flatten');
+const Logger = require('@cumulus/logger');
+const { Search, defaultIndexAlias } = require('./search');
+
+const log = new Logger({ sender: '@api/es/esFileSearchQueue' });
+
+const defaultESScrollSize = 1000;
+const defaultESScrollDuration = '30s';
+
+const logAndToss = (error) => {
+  log.error(JSON.stringify(error));
+  throw error;
+};
+
+/**
+ * Returns a function that returns truthyness value that an object has a key:bucket
+ * that matches the input `bucket`.
+ *
+ * @param {string} bucket - bucket to build a filter for.
+ * @returns {Function} - filter function
+ */
+const sameBucket = (bucket) => (object) =>
+  object.bucket && object.bucket === bucket;
+
+/**
+ * Given the Elasticsearch response, transform the array into an array of new
+ * objects where only files in the desired bucket are included and the
+ * granuleId is added.
+ *
+ * @param {Array<Object>} granuleFilesList - array of Elasticsearch results hits
+ * @param {string} bucket - bucket to include in results.
+ * @returns {Array<Object>}
+ */
+const buildFilesResponse = (granuleFilesList, bucket) =>
+  flatten(
+    granuleFilesList.map((gfl) =>
+      gfl.files.filter(sameBucket(bucket)).map((object) => ({
+        granuleId: gfl.granuleId,
+        ...object
+      })))
+  );
+
+class ESFileSearchQueue {
+  constructor({ bucket, esIndex }) {
+    this.items = [];
+    this.bucket = bucket;
+    this.params = {
+      index: esIndex || process.env.ES_INDEX || defaultIndexAlias,
+      type: 'granule',
+      size: process.env.ES_SCROLL_SIZE || defaultESScrollSize,
+      scroll: defaultESScrollDuration,
+      _source: ['files', 'granuleId'],
+      body: {
+        query: {
+          term: {
+            'files.bucket.keyword': {
+              value: `${bucket}`
+            }
+          }
+        },
+        sort: [
```

Comment:
Is this sorted by any file in the granule or the file with the specified bucket?

Reply:
I don't know. I thought it was for any file that was in the bucket, but I guess I'm not 100% sure.

Reply:
I tried the query in kibana, and it returned back record with `sort` field from a different bucket.
e.g. 
Query on `podaac-uat-cumulus-protected`
returns
`            {
              "bucket" : "podaac-uat-cumulus-public",
              "fileName" : "20190927000500-JPL-L2P_GHRSST-SSTskin-MODIS_A-D-v02.0-fv01.0.cmr.json",
              "size" : 1266,
              "type" : "metadata",
              "key" : "20190927000500-JPL-L2P_GHRSST-SSTskin-MODIS_A-D-v02.0-fv01.0.cmr.json"
            }
          ]
        },
        "sort" : [
          "20190927000500-JPL-L2P_GHRSST-SSTskin-MODIS_A-D-v02.0-fv01.0.cmr.json"
        ]`
It would be an issue if es just picks any file of the granule and sort them.

Reply:
There is `nested-sorting`, and I dtk if it helps.
https://www.elastic.co/guide/en/elasticsearch/reference/current/search-request-body.html#nested-sorting

Reply:
Yup, you're of course correct here.  Lauren and I are working on this, but sounds like it's going to take a new index.